lib/mongo/cluster/sdam_flow.rb

```diff
@@ -188,7 +188,6 @@ def server_description_changed
 
       verify_invariants
       commit_changes
-      disconnect_servers
```

Comment:
I'm not sure why we were disconnecting the servers on every server description change... This was causing a lint error because on server disconnect we clear the pool, and we would occasionally clear the pool for a known server.